tests/test_io.py

```diff
@@ -0,0 +1,426 @@
+            ProteinHelper.check_resnames({"ALA", "CYS", "XYZ"})
+
+        # Test with resnames not with custom template
+        with pytest.raises(
+            ValueError,
+            match=r"Residue \{'ALA'\} is not a standard residue or "
+            r"not in the templates\. Please provide a custom template\.",
+        ):
+            ProteinHelper.check_resnames(
+                {"ALA", "ABC", "XYZ"}, templates=[CUSTOM_TEMPLATE]
+            )
+
+    def test_n_residue_heavy_atoms(self, INPUT_MOL: Molecule) -> None:
+        """Test the counting of heavy atoms in residues."""
+        from prolif.io.protein_helper import ProteinHelper
+
+        # Test with a Molecule object
+        n_heavy_atoms = ProteinHelper.n_residue_heavy_atoms(INPUT_MOL.residues[1])
+        assert isinstance(n_heavy_atoms, int)
+        assert n_heavy_atoms == 11
+
+    def test_n_template_residue_heavy_atoms(self, CUSTOM_TEMPLATE: dict) -> None:
+        """Test the counting of heavy atoms in template residues."""
+        from prolif.io.protein_helper import ProteinHelper
+
+        # Test with a standard amino acid template
+        template_n_heavy_atoms = ProteinHelper.n_template_residue_heavy_atoms()
+        assert isinstance(template_n_heavy_atoms, dict)
+        assert template_n_heavy_atoms["ALA"] == 5
+
+        # Test with a custom (SMILES) template
+        custom_template_n_heavy_atoms = ProteinHelper.n_template_residue_heavy_atoms(
+            templates=[
+                CUSTOM_TEMPLATE,
+                {"XYZ": {"name": "XYZ", "test": "duplicate XYZ will be skiped."}},
+            ]
+        )
+        assert isinstance(custom_template_n_heavy_atoms, dict)
+        assert custom_template_n_heavy_atoms["XYZ"] == 6
+
+    def test_fix_molecule_bond_orders(
+        self, INPUT_MOL: Molecule, CUSTOM_TEMPLATE: dict
+    ) -> None:
+        """Test the fixing of bond orders in a Molecule object."""
+        from prolif.io.protein_helper import ProteinHelper
+
+        # Test with a Molecule object (using CIF template)
+        fixed_mol = ProteinHelper.fix_molecule_bond_orders(
+            INPUT_MOL.residues[1], templates=[CUSTOM_TEMPLATE]
+        )
+        assert isinstance(fixed_mol, Residue)
+
+        # Test with a Molecule object (using SMILES template)
+        fixed_mol_custom = ProteinHelper.fix_molecule_bond_orders(
+            INPUT_MOL.residues[1],
+            templates=[
+                {
+                    "TPO": {
+                        "name": "TPO",
+                        "SMILES": "C[C@H]([C@@H](C(=O))N)OP(=O)(O)O",
+                    }
+                }
+            ],
+        )
+        assert isinstance(fixed_mol_custom, Residue)
+
+        # check two residue are equal and bond orders are fixed
+        for at1, at2 in zip(
+            fixed_mol.GetAtoms(), fixed_mol_custom.GetAtoms(), strict=False
+        ):
+            assert at1.GetSymbol() == at2.GetSymbol()
+
+        bond1_info, bond2_info = [], []
```

Comment:
not a fan of this test, I'd just check that the number of bonds is the same, and then just iterate over the bonds of either, get the corresponding bond in the other mol, and check that the type is the same, something like:
```python
for bond1 in fixed_mol.GetBonds():
    bond2 = fixed_mol_custom.GetBondBetweenAtoms(bond1.GetBeginAtomIdx(), bond1.GetEndAtomIdx())
   assert bond1.GetBondType() is bond2.GetBondType()
```
in the end it does the same but is much simpler